src/Compilers/CSharp/Portable/Symbols/LocalDeclarationKind.cs

```diff
@@ -49,5 +49,16 @@ internal enum LocalDeclarationKind : byte
         /// The variable that captures the result of a pattern matching operation like "i" in "expr is int i"
         /// </summary>
         PatternVariable,
+
+        /// <summary>
+        /// User variable declared by a declaration expression in the left-hand-side of a deconstruction assignment.
+        /// </summary>
+        Deconstruction,
+
+        /// <summary>
+        /// User variable declared by a declaration expression that is not a deconstruction. This occurs for an
+        /// out variable declaration, and as a result of error recovery in incorrect code.
+        /// </summary>
+        DeclarationExpression,
```

Comment:
> it reflects the categorization that we have already found useful within the compiler

So far, there is only one place in the compiler that uses LocalDeclarationKind.PatternVariable in a meaningful way. And so far, we haven't found useful to have distinct kinds for Out Vars and Deconstruction vars. As I understand, you are claiming that we might take advantage of that in the future. I can buy that.    

> The purpose of this member is to expose rather than discard the information we already have

I am not proposing to discard any information, merely exposing an inconsistency in the proposed categorization. The main point I am trying to make is that, if we think that having a distinct kind for deconstruction variables can possibly benefit us in the future (clearly there are no benefits at the moment), then I believe it is reasonable to assume that we could benefit from distinct kind for Out variables (by comparison to variables that aren't Out variables) as well. 

So, if we are going towards increasing the "precision" of the kind, I am advocating for making one small extra step towards making it even more "precise", literally at almost no additional implementation cost. All it requires is a very simple syntax check in MakeDeclarationExpressionVariable.